pyfr/plugins/integrate.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+
+import re
+
+import numpy as np
+
+from pyfr.inifile import Inifile
+from pyfr.mpiutil import get_comm_rank_root, get_mpi
+from pyfr.plugins.base import BasePlugin, init_csv
+from pyfr.quadrules import get_quadrule
+from pyfr.nputil import npeval
+
+class IntegratePlugin(BasePlugin):
+    name = 'integrate'
+    systems = ['*']
+    formulations = ['dual', 'std']
+
+    def __init__(self, intg, cfgsect, suffix=None):
+        super().__init__(intg, cfgsect, suffix)
+
+        comm, rank, root = get_comm_rank_root()
+        
+        # Underlying elements class
+        self.elementscls = intg.system.elementscls
+
+        # Expressions to integrate
+        c = self.cfg.items_as('constants', float)
+        self.exprs = [(k, self.cfg.getexpr(cfgsect, k, subs=c))
+                      for k in self.cfg.items(cfgsect)
+                      if k.startswith('int-')]
+
+        # Gradient pre-processing
+        self._init_gradients(intg)
+
+        # Save a reference to the physical solution point locations
+        self.plocs = intg.system.ele_ploc_upts
+
+        # Integration parameters
+        self.nsteps = self.cfg.getint(cfgsect, 'nsteps')
+        
+        # The root rank needs to open the output file
+        if rank == root:
+            header = ["t"] + [(k) for k in self.cfg.items(cfgsect) if k.startswith('int-')]
+
+            # Open
+            self.outf = init_csv(self.cfg, cfgsect, ','.join(header))
+            
+        self.elminfo = []
+        # Iterate over the element types
+        for ename, eles in intg.system.ele_map.items():
+            # Locations of each solution point
+            plocupts = eles.ploc_at_np('upts')
+            
+            # Jacobians
+            rjacs = 1/eles.rcpdjac_at_np('upts')
+            
+            # Weights
+            rname = self.cfg.get('solver-elements-' + ename, 'soln-pts')
+            wts = get_quadrule(ename, rname, eles.nupts).wts
+            
+            # Save
+            self.elminfo.append((plocupts, rjacs, wts))
+        
```

Comment:
Remove newline.

Reply:
I think we can also pre-compute `wts/eles.rcpdjac_at_np('upts')` and just append this quotient to elminfo.